src/features/counterfactual/utils.ts

```diff
@@ -65,8 +67,11 @@ export const dispatchTxExecutionAndDeploySafe = async (
 
     const deploymentTx = await sdkUnchecked.wrapSafeTransactionIntoDeploymentBatch(signedTx, txOptions)
 
+    // We need to estimate the actual gasLimit after the user has signed since it is more accurate than what useDeployGasLimit returns
+    const gas = await signer.estimateGas({ data: deploymentTx.data, value: deploymentTx.value, to: deploymentTx.to })
```

Comment:
Should we compare this to the estimation and give a prompt / warning if the difference is more than e.g. 50k gas?

Reply:
I like the idea but this would only happen after the user has pressed Sign in their wallet. I am not sure how helpful it would be to see a warning somewhere in our UI at that point so I would leave it out for now and hope there are no substantial differences between those first transactions. Worst case we have the info text to warn the user about higher cost on execution.